elastic/kibana
# [Watcher] Refactor to use client from `RequestHandlerContext`

## Summary

No new functionality is introduced here. The refactor has three goals Watcher server-side request handlers:

1. Use ES client from context provided on the `RequestHandler` (to improve test-ability)
2. Update the types for the `RequestHandler` param, query and body on all routes
3. Separate handler registration from register function

To achieve this the following changes were made:

1. Removed `elasticsearchService` (setup service) from `RouteDependencies)
2. Used the `TypeOf` helper from `@kbn/config-schema` to get runtime and static types for routes
3. Extended the http request handler context with the watcher specific elasticsearch JS client per the instructions here: https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#declare-a-custom-scoped-service

## How to test

Run through all available actions in the UI (as I have done) to ensure that routes are still working correctly (which they should as no responses or handler logic has changed)

### For maintainers
